buf/registry/module/v1beta1/branch_service.proto

```diff
@@ -37,6 +39,24 @@ service BranchService {
   rpc ListBranches(ListBranchesRequest) returns (ListBranchesResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
+  // Push existing Commits onto Branches.
+  //
+  // This operation is atomic. Either all Commits are pushed to the referenced Branches in the order
+  // provided or an error is returned.
+  //
+  // This operation does not allow the caller to assert the state of Branches before pushing any
+  // Commits onto them. As a result, multiple concurrent PushCommits operations that target the same
+  // Branch will all succeed, and the latest Commit on the Branch will be determined by the last
+  // request that succeeded. To deterministically push Commits onto a Branch, use
+  // PatchBranchHistories instead.
+  rpc PushCommits(PushCommitsRequest) returns (PushCommitsResponse);
```

Comment:
I struggled to name these RPCs appropriately. Open to suggestions.

I also put this here instead of `BranchService` as I feel it's `Branch`-centric, but open to other suggestions.

Reply:
The `BranchService` relates to the `Branch` entity, the `CommitService` relates to the `Commit` entity. You are operating on a `Commit`, therefore it should be in the `CommitService`.

Reply:
My mental model is that I am operating on a `Branch` by putting a `Commit` on its history. The `Commit` is unaffected by this, nothing about it changes, but the `Branch` changes (it has a new latest commit).

Does that make sense, or am I thinking about this wrong?

Reply:
FWIW, my mental model of this is the same as Saquib's. This is an operation on a branch, not a commit.